sim/dex.js

```diff
@@ -521,6 +518,10 @@ class ModdedDex {
 		} else {
 			effect = new Data.PureEffect({name, exists: false});
 		}
+
+		if (effect) {
+			this.effectCache.set(name, effect);
+		}
 		// @ts-ignore
 		return effect;
```

Comment:
Also I think at this point, the rest of the ts-ignores can be casts.

```suggestion
		return /** @type {PureEffect} */ (effect);
```

Reply:
I was able to switch to a cast on 3/4 of the returns, but one complains:

`sim/dex.js:493:15 - error TS2352: Conversion of type 'Item | Move | Ability' to type 'PureEffect' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'Ability' is not comparable to type 'PureEffect'.
    Types of property 'effectType' are incompatible.
      Type '"Ability"' is not comparable to type '"Effect" | "Weather" | "Status"'.`